src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -47,23 +62,102 @@ public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options)
             }
         }
 
+        private static ConcurrentDictionary<string, ConsoleFormatter> SetFormatters(IEnumerable<ConsoleFormatter> fromSetup = null)
+        {
+            IEnumerable<ConsoleFormatter> formatters = fromSetup;
+            if (formatters == null || formatters.ToList().Count == 0)
+            {
+                var defaultMonitor = new FormatterOptionsMonitor<SimpleConsoleFormatterOptions>(new SimpleConsoleFormatterOptions());
+                var systemdMonitor = new FormatterOptionsMonitor<ConsoleFormatterOptions>(new ConsoleFormatterOptions());
+                formatters = new List<ConsoleFormatter>()
+                {
+                    new SimpleConsoleFormatter(defaultMonitor),
+                    new SystemdConsoleFormatter(systemdMonitor)
+                };
+            }
+            return new ConcurrentDictionary<string, ConsoleFormatter>(formatters.ToDictionary(f => f.Name));
+        }
+
+        // warning:  ReloadLoggerOptions can be called before the ctor completed,... before registering all of the state used in this method need to be initialized
         private void ReloadLoggerOptions(ConsoleLoggerOptions options)
         {
-            foreach (System.Collections.Generic.KeyValuePair<string, ConsoleLogger> logger in _loggers)
+            if (
+                options.FormatterName == null ||
+                !_formatters.TryGetValue(options.FormatterName, out ConsoleFormatter logFormatter) ||
+                !_formatters.TryGetValue(options.FormatterName?.ToLower(), out logFormatter)
```

Comment:
`ConcurrentDictionary` has a constructor that takes an IEqualityComparer for the keys. When you create the `_formatters` dictionary, you should just create it with an OrdinalIgnoreCase comparer, and then you won't have to ToLower or double-check the dictionary here. You can just call `_formatters.TryGetValue(options.FormatterName, out ConsoleFormatter logFormatter)`, and the right comparison will happen.